sktalk/corpus/parsing/eaf.py

```diff
@@ -0,0 +1,58 @@
+from itertools import chain
+from typing import Optional
+from pympi.Elan import Eaf
+from ..utterance import Utterance
+from .parser import InputFile
+
+
+class EafFile(InputFile):
+    """Parser for ELAN files."""
+
+    def __init__(self, path: str, tiers: Optional[list[str]] = None):
+        super().__init__(path)
+        self._tiers = [tiers] if isinstance(tiers, str) else tiers
+
+    def _extract_metadata(self):
+        return {
+            "header": self.pympi_eaf.header,
+            "adocument": self.pympi_eaf.adocument,
+            "licenses": self.pympi_eaf.licenses,
+            "locales": self.pympi_eaf.locales,
+            "languages": self.pympi_eaf.languages,
+            "media_descriptors": self.pympi_eaf.media_descriptors,
+            "properties": self.pympi_eaf.properties,
+            "linked_file_descriptors": self.pympi_eaf.linked_file_descriptors,
+            "constraints": self.pympi_eaf.constraints,
+            "linguistic_types": self.pympi_eaf.linguistic_types,
+            "controlled_vocabularies": self.pympi_eaf.controlled_vocabularies,
+            "external_refs": self.pympi_eaf.external_refs,
+            "lexicon_refs": self.pympi_eaf.lexicon_refs
+        }
+
+    def _extract_utterances(self):
+        available_tiers = self.pympi_eaf.tiers
+        if self._tiers:
+            tiers = [tier for tier in available_tiers if tier in self._tiers]
+            for tier in self._tiers:
+                if tier not in tiers:
+                    available = "; ".join(available_tiers.keys())
+                    raise KeyError(
+                        f"Tier {tier} not found in the file. Available tiers: {available}")
+        else:
+            tiers = available_tiers
+        utterances = [self._annotation_to_utterances(tier) for tier in tiers]
+        utterances = list(chain(*utterances))
```

Comment:
Perhaps do the flattening of the list on one line...
```suggestion
        utterances = [utterance for tier in tiers for utterance in self._annotation_to_utterances(tier)]
```